### ISSUE TITLE
Add typing to isbn3

### ISSUE THREAD

**Issue by doc-han** [inventaire/isbn3#8]
Consider adding types for isbn3 to allow usage with typescript. 
eg.

```typescript
interface ISBNAuditObject {
    source: string,
    validIsbn: boolean,
    groupname?: string,
    clues: Array<{
        message: string,
        candidate: string,
        groupname: string
    }>
}

interface ISBNObject {
    source: string,
    isValid: boolean,
    isIsbn10: boolean,
    isIsbn13: boolean,
    prefix?: string,
    group: string,
    publisher: string,
    article: string,
    check: string,
    isbn13?: string,
    isbn13h?: string,
    check10: string,
    check13: string,
    groupname: string,
    isbn10?: string,
    isbn10h?: string
}

declare module "isbn3" {
    function parse(isbn: string): ISBNObject | null;
    function asIsbn13(isbn: string): string;
    function asIsbn10(isbn: string): string;
    function hyphenate(isbn: string): string;
    function audit(isbn: string): ISBNAuditObject;
    const groups: Record<string, {
        name: string,
        ranges: Array<[string, string]>
    }>;

    export = { parse, asIsbn10, asIsbn13, hyphenate, audit, groups }
}
```

**Comment by anulman** [comment]
FWIW my linter complained about the `export =` line; would you consider using `export function ...` (and `export const ...`) in the `declare module 'isbn3' { ... }` block?

```ts
declare module "isbn3" {
  export function parse(isbn: string): ISBNObject | null;
  export function asIsbn13(isbn: string): string;
  export function asIsbn10(isbn: string): string;
  export function hyphenate(isbn: string): string;
  export function audit(isbn: string): ISBNAuditObject;
  export const groups: Record<
    string,
    {
      name: string;
      ranges: Array<[string, string]>;
    }
  >;
}
```

**Comment by maxlath** [comment]
Done in #11 and published in `v1.1.33`
